### PR TITLE
feat: Upgrade to crossplane-runtime `v1.14.4`

### PR DESCRIPTION
**Pull request by MisterMX**
### Description of your changes

Add support for store configs.
Add support for management policies.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

n.a. since no changes in the controller logic have been made. Only Setup has been adjusted.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -10,8 +10,8 @@ on:
 
 env:
   # Common versions
-  GO_VERSION: '1.20'
-  GOLANGCI_VERSION: 'v1.52.2'
+  GO_VERSION: '1.21.6'
+  GOLANGCI_VERSION: 'v1.54.0'
```
Comment (somebadcode) [inline .github/workflows/ci.yaml:14]: Shouldn't this be bumped to `v1.55.2`?

**Review by janwillies — CHANGES_REQUESTED** [review]
Thanks @MisterMX! Looks good to me, but before merging please run `s/aws/gitlab/`

File: apis/v1alpha1/doc.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package v1alpha1 contains core AWS resources.
+// +kubebuilder:object:generate=true
+// +groupName=aws.crossplane.io
```
Comment (janwillies) [inline apis/v1alpha1/doc.go:19]: These should be gitlab and a couple others too

**Review by janwillies — CHANGES_REQUESTED** [review]
Some of `aws` still present :)

**Review by janwillies — APPROVED** [review]
LGTM, thanks @MisterMX!

**Pull request merged**